x-pack/solutions/observability/packages/agent-builder/src/components/ai_insight_error_banner.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiCallOut, EuiButton } from '@elastic/eui';
+
+export interface AiInsightErrorBannerProps {
+  error?: string;
+  onRetry?: () => void;
+}
+
+export function AiInsightErrorBanner({ error, onRetry }: AiInsightErrorBannerProps) {
+  const title = i18n.translate('observabilityAgentBuilder.aiInsight.errorTitle', {
+    defaultMessage: 'Failed to generate AI insight',
+  });
+  const message = i18n.translate('observabilityAgentBuilder.aiInsight.errorMessage', {
+    defaultMessage: `The AI insight could not be generated due to the error: ${error}`,
+  });
+
+  return (
+    <EuiCallOut
+      announceOnMount
+      color="danger"
+      iconType="alert"
+      title={title}
+      data-test-subj={`AiInsightErrorBanner`}
+    >
+      <p>{message}</p>
```

Comment:
nit:
let's move the message here since it's not being resused:


```suggestion
      <p>{i18n.translate('observabilityAgentBuilder.aiInsight.errorMessage', {
    defaultMessage: `The AI insight could not be generated: ${error}`,
  })}</p>
```